engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessorMultipleProcessorsTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.streamprocessor;
+
+import static io.camunda.zeebe.engine.util.RecordToWrite.command;
+import static io.camunda.zeebe.engine.util.RecordToWrite.event;
+import static io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent.ACTIVATE_ELEMENT;
+import static io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent.ELEMENT_ACTIVATING;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.engine.api.EmptyProcessingResult;
+import io.camunda.zeebe.engine.api.RecordProcessor;
+import io.camunda.zeebe.engine.util.Records;
+import io.camunda.zeebe.engine.util.StreamPlatform;
+import io.camunda.zeebe.engine.util.StreamPlatformExtension;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceRecord;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InOrder;
+import org.mockito.verification.VerificationWithTimeout;
+
+@ExtendWith(StreamPlatformExtension.class)
+final class StreamProcessorMultipleProcessorsTest {
+
+  private static final long TIMEOUT_MILLIS = 2_000L;
+  private static final VerificationWithTimeout TIMEOUT = timeout(TIMEOUT_MILLIS);
+
+  private static final ProcessInstanceRecord RECORD = Records.processInstance(1);
+  private static final JobRecord JOB_RECORD = Records.job(1);
+
+  @SuppressWarnings("unused") // injected by the extension
+  private StreamPlatform streamPlatform;
+
+  @Test
+  void shouldChooseTheRightProcessorToProcess() {
+    // given
+    streamPlatform.writeBatch(command().processInstance(ACTIVATE_ELEMENT, RECORD).causedBy(0));
+    streamPlatform.writeBatch(command().job(JobIntent.COMPLETE, JOB_RECORD).causedBy(0));
+
+    final var processInstanceProcessor = createRecordProcessorFor(ValueType.PROCESS_INSTANCE);
+    final var jobProcessor = createRecordProcessorFor(ValueType.JOB);
+
+    // when
+    streamPlatform
+        .withRecordProcessors(List.of(processInstanceProcessor, jobProcessor))
+        .startStreamProcessor();
+
+    // then
+    final InOrder inOrder = inOrder(processInstanceProcessor, jobProcessor);
+    inOrder.verify(processInstanceProcessor, TIMEOUT).process(any(), any());
```

Comment:
:wrench: We could verify here that the record matches the RECORD to be more restrictive :)